src/keys.rs

```diff
@@ -16,11 +15,6 @@ pub fn keyval_name(keyval: u32) -> Option<GString> {
 pub fn keyval_to_unicode(keyval: u32) -> Option<char> {
     skip_assert_initialized!();
     unsafe {
-        let c: char = transmute(gdk_sys::gdk_keyval_to_unicode(keyval));
-        if c != '\0' {
-            Some(c)
-        } else {
-            None
-        }
+        ::std::char::from_u32(gdk_sys::gdk_keyval_to_unicode(keyval))
```

Comment:
Should I still check for `'\0'` in here too?

Reply:
Yes because that's what the function documentation says. It returns 0 if it didn't find a corresponding unicode character